Assets/Tests/Editor/NetworkWorldTests.cs

```diff
@@ -299,5 +298,28 @@ public void ClearDoesNotInvokeEvent()
             world.ClearSpawnedObjects();
             unspawnListener.DidNotReceiveWithAnyArgs().Invoke(default);
         }
+
+        [Test]
+        public void ClearDestoryedObjects()
+        {
+            AddValidIdentity(out uint id1, out NetworkIdentity identity1);
+            AddValidIdentity(out uint id2, out NetworkIdentity identity2);
+            AddValidIdentity(out uint id3, out NetworkIdentity identity3);
+
+            NetworkIdentity nullIdentity = new GameObject("WorldTest").AddComponent<NetworkIdentity>();
+            nullIdentity.NetId = getValidId();
+            world.AddIdentity(nullIdentity.NetId, nullIdentity);
+
+            Object.DestroyImmediate(nullIdentity);
+
+            Assert.That(world.SpawnedIdentities.Count, Is.EqualTo(4));
+
+            world.RemoveDestroyedObjects();
```

Comment:
we should check the Count is 3 after removing the destroyed object

Reply:
good point